File: mario.py
def main():
    while True:
        i = int(input("height of wall: "))
        if i > 0 and i <= 23:
            break 

# print out this many rows
    for s in range(i):
        
        for a in range(i):
            if s + 1 + a < i:
                print(" ", end="")
            else:
                print("#", end="")
        print()
if __name__== "__main__":
    main()

File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    do
    {
        printf("heights of wall:");
        height = get_int();
    }
    while ((height < 0) || (height > 23));

    for (int i = 1; i <= height ; i++)
    {
        for (int b = 0; b <= height ; b++)
        {
            if ((i + b) < height)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}